kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/AbstractFilesArtifactResolver.java

```diff
@@ -99,7 +99,12 @@ private boolean isInJarStructuredFolder(String name, String type) {
     }
 
     private PomParser buildPomParser(AFReleaseId releaseId) {
-        List<URL> url = effectivePoms.stream().filter(e -> e.getFile().endsWith(toFile(releaseId, "pom"))).collect(toList());
+        final String pomName = toFile(releaseId, "pom");
+        List<URL> url = effectivePoms.stream().filter(e -> e.getFile().endsWith(pomName)).collect(toList());
+        if (url.isEmpty() && pomName.contains("-SNAPSHOT")) {
```

Comment:
Here I would check if SNAPSHOT is on version
So the comparison will be `releaseId.getVersion().endWith("-SNAPSHOT")` and then
```
 url = effectivePoms.stream().filter(e -> e.getFile().find(releaseId.getArtifactId()+"-"+releaseId.getVersion().replace("-SNAPSHOT",".*") ++".pom$")).collect(toList());
```
you do not need the final variable, $ denotes end of string,  therefore you do not need to append .* at the beggining of the regexpr and you can should use  find rather than match

Reply:
yes, that looks better and limit the regex. Changing it